### PR TITLE
Add docstrings and doc deployment with Documenter

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

File: src/AWS4AuthRequest.jl [hunk 1]
```diff
@@ -5,30 +5,37 @@ using Dates
...
+[AWS Signature v4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html)
+authentication layer to the request.
+"""
+abstract type AWS4AuthLayer{Next<:Layer} <: Layer end
 
-Credentials are read from environment variables `AWS_ACCESS_KEY_ID`,
-`AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`.
 """
-abstract type AWS4AuthLayer{Next <: Layer} <: Layer end
+    HTTP.request(AWS4AuthLayer, url::HTTP.URI, req::HTTP.Request, body) -> HTTP.Response
```
Comment (omus) [inline src/AWS4AuthRequest.jl]: `AWS4AuthLayer` isn't clearly a type here

File: src/AWS4AuthRequest.jl [hunk 2]
```diff
@@ -5,30 +5,37 @@ using Dates
...
+
+Perform the given request, adding a layer of AWS authentication using
+[AWS Signature v4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
+An "Authorization" header to the request.
 
-function HTTP.request(::Type{AWS4AuthLayer{Next}}, url::URI, req, body; kw...) where Next
+Credentials used for the signing are read from environment variables `AWS_ACCESS_KEY_ID`,
+`AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN`.
```
Comment (omus) [inline src/AWS4AuthRequest.jl]: Credentials could also be read from a file or be instance based. May be best to just be less specific

Comment (ararslan) [reply src/AWS4AuthRequest.jl]: I've just deleted the mention of credentials here.

File: src/AWS4AuthRequest.jl [hunk 3]
```diff
@@ -44,6 +48,30 @@ function _normalize_ws(s::AbstractString)
     end
 end
 
+"""
+    sign_aws4!(method::String, url::HTTP.URI, headers::HTTP.Headers, body; kwargs...)
+
+Add an "Authorization" header to `headers`, modifying it in place.
+The header contains a computed signature based on the given credentials as well as
+metadata about what was used to compute the signature.
+For more information, see the AWS documentation on the Signature v4 process.
```
Comment (omus) [inline src/AWS4AuthRequest.jl]: Let's just make this a link too

**Review by omus — DISMISSED** [review]

**Review by omus — APPROVED** [review]

**Pull request merged**